index.html

```diff
@@ -10314,6 +10314,33 @@ <h3><dfn>Release Actions</dfn></h3>
 as if the state was released by an explicit series of actions.
 It also clears all the internal state of the virtual devices.
 
+<p>To <dfn>get undo actions</dfn> given <var>input
+ state</var>:
+
+<ol class=algorithm>
+ <li><p>Let <var>token</var> be a new unique identifier.
+
+ <li><p>Enqueue <var>token</var> in <var>input state</var>&apos;s <a>actions
+  queue</a>.
+
+ <li><p>Wait for <var>token</var> to be the first item
+ in <var>input state</var>&apos;s <a>actions queue</a>.
+
+ <aside class=note>
+   <p>This ensures that only one set of actions can be run at a time,
+   and therefore different actions commands using the same underlying
+   state don&apos;t race. In a session that is only a HTTP session only one
+   command can run at a time, so this will never block. But other
+   session types can allow running multiple commands in parallel, in
+   which case this is necessary to ensure sequential access.
+ </aside>
+
```

Comment:
@jgraham this is a bit of duplication but I hope it's fine. Otherwise we might have to create a new algorithm, that is just waiting for the token to be first in the queue.

Reply:
I think factoring this out into a new algorithm would be better.